src/Symfony/Component/AutoMapper/Generator/Generator.php

```diff
@@ -0,0 +1,430 @@
+                    $conditions[] = new Expr\FuncCall(new Name('array_key_exists'), [
+                        new Arg(new Scalar\String_($propertyMapping->getProperty())),
+                        new Arg($sourceInput),
+                    ]);
+                }
+            }
+
+            if ($mapperGeneratorMetadata->shouldCheckAttributes()) {
+                $conditions[] = new Expr\MethodCall($contextVariable, 'isAllowedAttribute', [
+                    new Arg(new Scalar\String_($propertyMapping->getProperty())),
+                ]);
+            }
+
+            if (null !== $propertyMapping->getSourceGroups()) {
+                $conditions[] = new Expr\BinaryOp\BooleanAnd(
+                    new Expr\BinaryOp\NotIdentical(
+                        new Expr\ConstFetch(new Name('null')),
+                        new Expr\MethodCall(new Expr\Variable('context'), 'getGroups')
+                    ),
+                    new Expr\FuncCall(new Name('array_intersect'), [
+                        new Arg(new Expr\MethodCall(new Expr\Variable('context'), 'getGroups')),
+                        new Arg(new Expr\Array_(array_map(function (string $group) {
+                            return new Expr\ArrayItem(new Scalar\String_($group));
+                        }, $propertyMapping->getSourceGroups()))),
+                    ])
+                );
+            }
+
+            if (null !== $propertyMapping->getTargetGroups()) {
+                $conditions[] = new Expr\BinaryOp\BooleanAnd(
+                    new Expr\BinaryOp\NotIdentical(
+                        new Expr\ConstFetch(new Name('null')),
+                        new Expr\MethodCall(new Expr\Variable('context'), 'getGroups')
+                    ),
+                    new Expr\FuncCall(new Name('array_intersect'), [
+                        new Arg(new Expr\MethodCall(new Expr\Variable('context'), 'getGroups')),
+                        new Arg(new Expr\Array_(array_map(function (string $group) {
+                            return new Expr\ArrayItem(new Scalar\String_($group));
+                        }, $propertyMapping->getTargetGroups()))),
+                    ])
+                );
+            }
+
+            if (null !== $propertyMapping->getMaxDepth()) {
+                $conditions[] = new Expr\BinaryOp\SmallerOrEqual(
+                    new Expr\MethodCall($contextVariable, 'getDepth'),
+                    new Scalar\LNumber($propertyMapping->getMaxDepth())
+                );
+            }
+
+            if (\count($conditions) > 0) {
+                $condition = array_shift($conditions);
+
+                while (\count($conditions) > 0) {
+                    $condition = new Expr\BinaryOp\BooleanAnd($condition, array_shift($conditions));
+                }
+
+                $propStatements = [new Stmt\If_($condition, [
+                    'stmts' => $propStatements,
+                ])];
+            }
+
+            $statements = array_merge(
```

Comment:
The call to `array_merge` should be moved outside of the loop: https://github.com/dseguy/clearPHP/blob/master/rules/no-array_merge-in-loop.md#no-array_merge-in-loops

Reply:
Replaced this with a foreach loop instead of array_merging outside the loop, it keeps the same behavior and performance and remove dealing with extra variables

Recently got this use case in another project and find this the better solution